src/invidious/routes/video_playback.cr

```diff
@@ -294,6 +294,9 @@ module Invidious::Routes::VideoPlayback
     end
 
     if local
+      if (CONFIG.invidious_companion)
+        return env.redirect "#{video.invidious_companion["baseUrl"].as_s}#{env.request.path}?#{env.request.query}"
+      end
```

Comment:
Here are the specific reasons which are mostly the same from dash api:
- One may configure a proxy in invidious_companion. And invidious companion is going to handle multiple proxies soon, so this can't be handled inside Invidious: https://github.com/iv-org/invidious-companion/issues/5
- One may configure multiple invidious companion, we need to make sure that the video stream used is from the same public IP address that generated the video stream.